src/OpenTK/Platform/X11/XI2MouseKeyboard.cs

```diff
@@ -167,11 +167,32 @@ internal static bool IsSupported(IntPtr display)
             return false;
         }
 
+        private bool GetIndicatorState(string name)
+        {
+            bool state = false;
+            bool lockState = false;
+            int ndx = 0;
+            IntPtr map = new IntPtr ();
+            bool real = false;
+        
+            IntPtr atom = Functions.XInternAtom(API.DefaultDisplay, name, true);
+            Xkb.GetNamedIndicator(API.DefaultDisplay, atom, out ndx, out state, out map, out real);
+            
+            return state;
+        }
+
         KeyboardState IKeyboardDriver2.GetState()
         {
             lock (Sync)
             {
                 KeyboardState state = new KeyboardState();
+
+                if (Xkb.IsSupported (API.DefaultDisplay))
+                {
+                    state.CapsLock = GetIndicatorState("Caps Lock");
+                    state.NumLock = GetIndicatorState("Num Lock");                     
+                }
+
```

Comment:
Similar to the Windows comment, can this be done in ProcessRawEvent instead of GetState.